lib/klaviyo/client.rb

```diff
@@ -1,24 +1,112 @@
 require 'open-uri'
 require 'base64'
 require 'json'
+#require pry for testing
+require 'pry'
 
 module Klaviyo
   class KlaviyoError < StandardError; end
-  
+
   class Client
-    def initialize(api_key, url = 'http://a.klaviyo.com/')
+    def initialize(api_key = nil, private_api_key = nil, url = 'https://a.klaviyo.com/api/')
+      if !api_key
+        KlaviyoError.new('Please provide your Public API key')
+      end
+
       @api_key = api_key
+      @private_api_key = private_api_key
       @url = url
+
+      if @private_api_key
+        @private_api_key_param = "api_key=#{@private_api_key}"
+      end
+# API versions
+      @metrics_version = 'v1'
+
+# API endpoints
+      @metrics_path = "#{@metrics_version}/metrics"
+      @timeline_path = 'timeline'
+      @metrics_timeline_path = "#{@metrics_path}/#{@timeline_path}"
+      @metric_timeline_path = "#{@metrics_version}/metric"
```

Comment:
do you need to store all these constructed endpoints as instance variables or can you just create them as you use them?